Client/Frontend/Browser/BrowserViewController/BrowserViewController+TabToolbarDelegate.swift

```diff
@@ -75,6 +75,8 @@ extension BrowserViewController: TabToolbarDelegate, PhotonActionSheetProtocol {
         let isPrivate = tabManager.selectedTab?.isPrivate ?? false
         tabManager.selectTab(tabManager.addTab(nil, isPrivate: isPrivate))
         focusLocationTextField(forTab: tabManager.selectedTab)
+        overlayManager.openNewTab(url: nil,
+                                  newTabSettings: NewTabAccessors.getNewTabPage(profile.prefs))
```

Comment:
I'm trying to understand why we need to add a call to the `overlayManager` in this particular PR to remove the legacy store? I guess this relates to the tests that were failing right?

Reply:
Yup, for some reason it no longer triggers overlay mode at the right time. There is likely a superfluous call at the wrong time somewhere else but I was afraid to start messing with it because it's spagetified into other code paths.